addon/components/frost-table.js

```diff
@@ -54,12 +81,34 @@ export default Component.extend({
     return columns.map((column, index) => Object.assign({index}, column))
   },
 
+  @readOnly
+  @computed()
+  _isSelectable () {
+    return !isNone(this.get('onSelectionChange'))
+  },
+
   // == Functions =============================================================
 
+  setShift (event) {
+    if (!this.isDestroyed) {
+      this.set('_isShiftDown', event.shiftKey)
+    }
+  },
+
   // == DOM Events ============================================================
 
   // == Lifecycle Hooks =======================================================
 
+  init () {
+    this._super(...arguments)
+    const itemKey = this.get('itemKey')
+    if (itemKey) {
+      this.set('_itemComparator', function (lhs, rhs) {
```

Comment:
Nope - different purposes - the CP in the body is to handle optimization of the `each` if no itemKey is provided.  This case is present only for selection/expansion, so it's actually two different cases